From e5c1d71344aeca339e37616cc8a5818c520c44a0 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 7 Nov 2017 17:18:44 +0800
Subject: [PATCH 2812/5242] MLK-16765-3 rtc: rtc-imx-sc: check group interrupt
 type

commit  27dc5decdf3f7e7aa397d1fa75926a2cf9c401af from
https://source.codeaurora.org/external/imx/linux-imx.git

Check group interrupt type and irq status type supported, return directlly
if it's not the right group and interrupt status.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/rtc/rtc-imx-sc.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-imx-sc.c b/drivers/rtc/rtc-imx-sc.c
index 4fd2619..9f09409 100644
--- a/drivers/rtc/rtc-imx-sc.c
+++ b/drivers/rtc/rtc-imx-sc.c
@@ -21,6 +21,7 @@
 #include <linux/rtc.h>
 #include <soc/imx/fsl_sip.h>
 #include <soc/imx8/sc/sci.h>
+#include <soc/imx8/sc/svc/irq/api.h>
 
 sc_ipc_t timer_ipcHandle;
 
@@ -31,10 +32,16 @@ struct imx_sc_rtc_data {
 
 struct imx_sc_rtc_data *data;
 
-static int imx_sc_rtc_alarm_sc_notify(struct notifier_block *nb, unsigned long event, void *dummy)
+static int imx_sc_rtc_alarm_sc_notify(struct notifier_block *nb,
+					unsigned long event, void *group)
 {
 	u32 events = 0;
 
+	/* ignore non-rtc irq */
+	if (!((event & SC_IRQ_RTC) &&
+		(*(sc_irq_group_t *)group == SC_IRQ_GROUP_RTC)))
+		return 0;
+
 	rtc_update_irq(data->rtc, 1, events);
 
 	return 0;
-- 
1.7.9.5

